toc/rfc/rfc-draft-ipv6-dual-stack-for-cf.md

```diff
@@ -0,0 +1,200 @@
+
+For supporting dual stack networking, the BOSH agent must support the following:
+
+1. Allow the agent to accept having multiple network settings for the same network
+
+   The BOSH Agent has a limitation that allows only a single BOSH network to be assigned to a Network Interface Card (NIC) in a virtual machine. When multiple BOSH networks are defined for the same hyperscaler subnet,
+   This limitation [has been resolved with bosh-agent#344 already.](https://github.com/cloudfoundry/bosh-agent/pull/344)
+
+2. With the BOSH CPI being able to delegate IPv6 prefixes to single VMs, the BOSH agent MUST be able to assign such CIDRs to VMs or at least not interfere with the setup (e.g. https://github.com/cloudfoundry/bosh-agent/pull/347 ), if the target infrastructure allows.
+
+#### BOSH Stemcells
+
+BOSH stemcells up to Ubuntu Jammy Jellyfish have IPv6 support disabled at kernel level. While it can be enabled, the resulting configuration change requires a VM reboot.
+
+The goal is to keep the already enabled IPv6 support from Ubuntu upstream enabled.
+This will be the case starting with Ubuntu Noble stemcell. Enabling IPv6 in older stemcell releases such as “Ubuntu Jammy” might be seen as backwards incompatible change.
+
+This topic is being discussed and tracked via the [issue os-conf-release#71](https://github.com/cloudfoundry/os-conf-release/issues/71) and must be followed up with appropriate fixes.
+
+### Diego
+
+Currently, the Silk CNI is used to create an overlay network, but also to assign IP addresses for the application containers. This assignment is done from a centrally managed pool of IP addresses.
+
+The goal of having distinct IPv6 addresses for the various containers can be achieved without such a central assignment. A Linux machine can receive a CIDR range on its network interface, e.g. `/80` or smaller, and receive traffic for any address within that range.
+
+The delegation of IPv6 addresses within the cells is managed through the host-local CNI plugin. Each cell is assigned an IPv6 prefix at the configuration level, which is then provided to the host-local IPAM plugin. Using this prefix, the plugin allocates unique IPv6 addresses to individual containers within the cell. To prevent address conflicts and ensure proper tracking, a state file is maintained at the cell level, keeping a record of all assigned addresses.
+
+This single machine, i.e. a Diego cell, then has sole control over this network range and can use it to assign specific IP addresses to its containers. At a network level, traffic is still sent to the Diego cell, and within the Diego cell can be forwarded to the application container based on its IP address directly as part of the networking stack.
+
+Network traffic from and to application containers needs to be controlled, also for IPv6:
+
+1. For inter-container communication, Application Security Groups (ASGs) need to support IPv6.
```

Comment:
I think majority of work would be to support dual-stack. For CNI wrapper plugin which configures ASGs IP address is an arbitrary string.

Reply:
I've changed the wording to "need to add support for IPv6", which implies dual stack. Everything in this RFC is about adding IPv6 to the existing IPv4 and thus reaching dual-stack support.

Do you think we need to be more explicit here?